name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/wizards/datatransfer/ReviewExtractedItemsPage.java

```diff
@@ -746,6 +747,34 @@ private void checkEntriesAndRefresh(List<ExtractedEntry> entries)
 
     private void checkEntries(List<ExtractedEntry> entries)
     {
+        // update dynamic currency if any
+        for (ExtractedEntry entry : entries)
```

Comment:
If we update the transactions at this point of time, then the currency is set to the first account selected. If the user selects a different account, then the transactions will not be updated again. So instead of the currency of the file, it now depends on the currency of the first account.

Instead, my proposal is:
* Extend the ```CheckCurrencyAction``` to accept transactions with the ```UNDEFINED``` currency **if** the transactions are okay if they would have the currency of the account.
* Extend the ```InsertAction``` to update the currency to the currency of the account if the currency code is ```UNDEFINED```

Reply:
I don't believe this is true.

checkEntries gets called when the account selection changes on the review page.
maybeUpdateDynamicCurrency gets the target currency from getAccount....which gets it from the currently selected account, not the first.

So it works, but here's the problem. If I don't store somewhere that it was "undefined" (this isDynamicCurrencyCode in the Model) then the first selection will override, and subsequent account selection changes will be stuck at the first account's currency.

I did actually go down the path of modifying the CheckCurrencyAction before, but:
 - I found there were at least 4 or 5 places that needed patching up, and it wasn't always clear
 - I was also a bit worried to give the UI check different values to when the import finally runs....it seemed cleaner to me to run through the checks with the currency set, so it can properly do it's thing.

How about I just put the is isDynamicCurrencyCode field into the reviewpage itself, it will get set on first load based on the first entries currencyCode == "UNDEFINED". Then the model doesn't have to change, the UI updates nicely with the currency, and we don't have to run preview checks differently from the final input checks?

Reply:
> How about I just put the is isDynamicCurrencyCode field into the reviewpage itself

That is a good idea!

Today, every extracted transaction is already wrapped in to a [Extractor.Item](https://github.com/buchen/portfolio/blob/a8eabbbe3b91f448c93f07e1021a50d6965c5a37/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/Extractor.java#L52). This item already has a generic ```data``` field to store the JSON representation (if a JSON-based extractor is used).

What do you think about this algorithm?
* The extractor creates a valid transactions - also with a valid currency (just to make sure we do not pollute with incorrect currencies if the UI is not used)
* The extractor tells the ```Extractor.Item``` that the currency is dynamic / undefined - I think a new boolean property to the ```Extractor.Item``` does not hurt
* Your logic reads this property and adopts the currency as needed *before* passing it to CheckCurrencyAction and InsertAction

Reply:
@chrisaut - just a quick check-in. Are you planning to update the change?

Reply:
Yes, sorry for the delay. I just didn't get to much of anything over the holidays. I should have some more time next weekend to look at all my open tasks for PP, if that's ok?